pkg/engine/armresources.go

```diff
@@ -47,13 +47,13 @@ func GenerateARMResources(cs *api.ContainerService) []interface{} {
 		armResources = append(armResources, userAssignedID, msiRoleAssignment)
 	}
 
-	if !cs.Properties.OrchestratorProfile.IsPrivateCluster() &&
-		!isHostedMaster &&
+	if !isHostedMaster &&
```

Comment:
This change creates the Standard LB for node pools in a private cluster scenario. Prior to this PR that LB was not being created in a private cluster scenario.

Reply:
maybe, 

```go
// IsNotAKSAndRequiresASLB should have a solid comment
if cs.Properties.IsNotAKSAndRequiresASLB() {
// ...
}
```
Secondarily, I think this is wrong `!cs.Properties.AnyAgentHasLoadBalancerBackendAddressPoolIDs()`. If there are multiple agent pools and only one of them has a backend pool ID, then we wouldn't make a load balancer and the other agent pools, besides the one that has a backend pool specified, would not be exposed to traffic.